smithy-model/src/main/java/software/amazon/smithy/model/knowledge/MemberPropertyIndex.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.smithy.model.knowledge;
+
+import java.util.HashMap;
+import java.util.Map;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.MemberShape;
+import software.amazon.smithy.model.shapes.ShapeId;
+import software.amazon.smithy.model.traits.NotPropertyTrait;
+
+/**
+ * Index of structure member -> property inclusion.
+ **/
+public final class MemberPropertyIndex implements KnowledgeIndex {
```

Comment:
I intentionally avoided that name because the relevance of the index is for callers scanning, iterating, and filtering through shape members aligned with top-level resource properties. Most importantly this index won't give a caller _all of the properties_ without being anchored in a particular operation. If that information was offered conveniently or would make sense to offer in this index then that names makes more sense.

Reply:
I think that properties are inherently a resource thing so it makes sense to be in the name. I don't think the distinction of needing to be anchored to an operation matters much. I'd also assume you really need resource + operation (I can't see the PR since it lists 1,100+ changes files right now).

Reply:
The name we're going with is `PropertyBindingIndex` following the pattern of `IdentifierBindingIndex`